osu.Game.Tests/Visual/Menus/TestSceneLoginPanel.cs

```diff
@@ -51,5 +55,22 @@ public void TestLoginFailure()
             AddStep("enter password", () => loginPanel.ChildrenOfType<OsuPasswordTextBox>().First().Text = "password");
             AddStep("submit", () => loginPanel.ChildrenOfType<OsuButton>().First(b => b.Text.ToString() == "Sign in").TriggerClick());
         }
+
+        [Test]
+        public void TestClickingOnFlagClosesPanel()
```

Comment:
nit: I would've put this in `UserPanel` instead so that it looks like it's testing whether `UserPanel.Action` will be invoked if the country drawable was clicked, but either way works anyways.

Reply:
disagree. imo it's better to test actual end user scenarios and behaviours rather than nebulous behaviours of abstract components devoid of context

Reply:
fair enough.